### ISSUE TITLE
Parallel Routes+Server Actions+Redirect+Turbopack doesn't works together

### ISSUE THREAD

**Issue by Altro31** [vercel/next.js#59445]
### Link to the code that reproduces this issue

https://codesandbox.io/p/devbox/test-for-parallel-routes-server-actions-redirect-6t2c69

### To Reproduce

1. Start application in development with turbopack <code>next dev --turbo</code>
2. Press on **_Test Parallel_**.
3. Press on **_Submitl_**
4. After that the whole app chashes until refresh the page. (**_Back_** don't works)

### Current vs. Expected behavior

When navigate to **Test Parallel** and press in Submit, it call the action pased to the form and only redirect to the "/" page, but  the whole page crashes instead and any other link doesn't works. The correct behavior, i think, is , aftter submit, redirects to "/" page.

### Verify canary release

- [x] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
  Platform: win32
  Arch: x64
  Version: Windows 11 Pro
Binaries:
  Node: 20.9.0
  npm: N/A
  Yarn: N/A
  pnpm: N/A
Relevant Packages:
  next: 14.0.5-canary.4
  eslint-config-next: 14.0.2
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.3.3
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

App Router, Routing (next/router, next/navigation, next/link), Turbopack (--turbo)

<sub>[PACK-2137](https://linear.app/vercel/issue/PACK-2137/parallel-routesserver-actionsredirectturbopack-doesnt-works-together)</sub>

**Comment by anptrs** [comment]
Yeap, I have the same issues. It seems the whole parallel routes, router cache, useRouter, <Link> and redirect is a hot mess at the moment.

The way I worked around the server's redirect not working half the time: I return a client RedirectOnLoad component intead of calling redirect(). The component then does the router.replace() at mount (after a small timeout) on the client. Timeout here is necessary or the router.replace() or router.push() does not work right after mount. Like I said this whole thing is a huge mess.

Also, using <a> instead of <Link> for parallel routs, works to side step a client side router cache, which is just broken at the moment.

**Comment by ForsakenHarmony** [comment]
I can't reproduce this on the latest canary, is this still an issue for you?

**Comment by Altro31** [comment]
This issue was resolved since version 14.0.4-canary.64
